test/gtest/gpu_mha_backward.cpp

```diff
@@ -178,12 +179,13 @@ class Test_Bwd_Mha : public testing::TestWithParam<TestCase>
             tensors[id] = std::move(tmp);
         };
 
-        auto GenScaledTensor = [](auto... nhsd) {
-            auto val_full = tensor<float>{nhsd...};
+        auto GenScaledTensor = [](auto type, auto... nhsd) {
```

Comment:
consider combining with `GetScaledTensor` and `ScaledTensor` in `mha_helper.hpp` as it will help with writing `graph API` tests for backward MHA.

Reply:
I noticed that it looked different. May be we should have two functions, each of which generates a `ScaledTensor` object in different ways.

Reply:
#3040